packages/video_player/video_player/ios/Classes/FLTVideoPlayerPlugin.m

```diff
@@ -451,10 +451,7 @@ - (void)detachFromEngineForRegistrar:(NSObject<FlutterPluginRegistrar>*)registra
     [player disposeSansEventChannel];
   }
   [_players removeAllObjects];
-  // TODO(57151): This should be commented out when 57151's fix lands on stable.
-  // This is the correct behavior we never did it in the past and the engine
-  // doesn't currently support it.
-  // FLTVideoPlayerApiSetup(registrar.messenger, nil);
+  FLTVideoPlayerApiSetup(registrar.messenger, nil);
```

Comment:
It doesn't look like 57151 landed on stable yet

Reply:
So [2759](https://github.com/flutter/plugins/pull/2759) isn't an appropriate workaround? Since 57151 is on master but not stable, would conditionally setting the messageHandler to nil if not on stable work (i.e. calling `flutter channel` using `NSTask` and storing the active channel in an `NSString` to make this check)?